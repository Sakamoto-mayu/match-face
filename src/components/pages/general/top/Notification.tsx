import { twMerge } from 'tailwind-merge';

type MenuProps = {
  title: string;
  completedTasksNumbers: number;
  incompletedTasksNumbers: number;
  deadlineDay: string;
  deadlineNumbers: number;
  className: string;
};

export default function Notification({
  title,
  completedTasksNumbers,
  incompletedTasksNumbers,
  deadlineDay,
  deadlineNumbers,
  className,
}: MenuProps) {
  //buttonのclassNameは親コンポーネントでclassNameを指定して上書き可能
  const mergedClassName = twMerge('p-5 w-1/2 h-auto shadow-lg', className);
  return (
    <>
      <div className={mergedClassName}>
        <h1 className="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12">
          {title}
        </h1>

        <div className="">
          {/* answer_statusがtrueの件数? */}
          <h2 className="">【対応済】{completedTasksNumbers}件</h2>
          {/* answer_statusがfalseの件数? */}
          <h2 className="pt-4">【未対応】{incompletedTasksNumbers}件</h2>
          {/* データ構造がはっきりしてから編集要*/}
          <ul className=" pl-4">
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
          </ul>
        </div>
      </div>
    </>
  );
}
