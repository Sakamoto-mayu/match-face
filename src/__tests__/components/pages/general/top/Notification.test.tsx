import Notification from '@/components/pages/general/top/Notification';
import { render } from '@testing-library/react';

type AnswerRequest = {
  id: number;
  user_id: number;
  administorator_id: number;
  project_id: number;
  request_at: string;
  deadline: string;
  answered: boolean;
  created_user: string;
  created_at: string;
  update_user: string;
  update_at: string;
};

type AnswerRequests = AnswerRequest[];

// answered === falseのデータ
export const falseAnswerRequests: AnswerRequests = [
  {
    id: 1,
    user_id: 1,
    administorator_id: 1,
    project_id: 1,
    request_at: '2023-09-11 12:14:38.347441',
    deadline: '2023-10-01 18:00:00',
    answered: false,
    created_user: 'ラクス責任者',
    created_at: '2023-09-11 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-09-11 12:14:38.347441',
  },
  {
    id: 2,
    user_id: 1,
    administorator_id: 1,
    project_id: 2,
    request_at: '2023-09-11 12:14:38.347441',
    deadline: '2023-10-15 18:00:00',
    answered: false,
    created_user: 'ラクス責任者',
    created_at: '2023-09-11 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-09-11 12:14:38.347441',
  },
  {
    id: 3,
    user_id: 1,
    administorator_id: 1,
    project_id: 3,
    request_at: '2023-09-11 12:14:38.347441',
    deadline: '2023-10-01 18:00:00',
    answered: false,
    created_user: 'ラクス責任者',
    created_at: '2023-09-11 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-09-11 12:14:38.347441',
  },
  {
    id: 4,
    user_id: 1,
    administorator_id: 1,
    project_id: 4,
    request_at: '2023-09-11 12:14:38.347441',
    deadline: '2023-12-15 18:00:00',
    answered: false,
    created_user: 'ラクス責任者',
    created_at: '2023-09-11 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-09-11 12:14:38.347441',
  },
];
// answered === trueのデータ
export const trueAnswerRequests: AnswerRequests = [
  {
    id: 5,
    user_id: 1,
    administorator_id: 1,
    project_id: 8,
    request_at: '2023-06-01 12:14:38.347441',
    deadline: '2023-06-30 18:00:00',
    answered: true,
    created_user: 'ラクス責任者',
    created_at: '2023-06-01 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-06-01 12:14:38.347441',
  },
  {
    id: 6,
    user_id: 1,
    administorator_id: 1,
    project_id: 9,
    request_at: '2023-06-01 12:14:38.347441',
    deadline: '2023-06-30 18:00:00',
    answered: true,
    created_user: 'ラクス責任者',
    created_at: '2023-06-01 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-06-01 12:14:38.347441',
  },
  {
    id: 7,
    user_id: 1,
    administorator_id: 1,
    project_id: 10,
    request_at: '2023-06-01 12:14:38.347441',
    deadline: '2023-08-01 18:00:00',
    answered: true,
    created_user: 'ラクス責任者',
    created_at: '2023-06-01 12:14:38.347441',
    update_user: 'ラクス責任者',
    update_at: '2023-06-01 12:14:38.347441',
  },
];

describe('スナップショット', () => {
  test('スナップショット', () => {
    const { container } = render(
      <Notification
        answeredAnswerRequests={trueAnswerRequests}
        notAnsweredAnswerRequests={falseAnswerRequests}
        className=""
      />
    );
    expect(container).toMatchSnapshot();
  });
});
