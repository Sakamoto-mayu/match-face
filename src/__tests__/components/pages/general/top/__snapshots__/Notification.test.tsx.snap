// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット スナップショット 1`] = `
<div>
  <div
    class="p-5 w-3/5 h-auto shadow-lg"
  >
    <h1
      class="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12"
    >
      進捗状況
    </h1>
    <div
      class="pt-5"
    >
      <h2>
        【対応済】
        3
        件
      </h2>
      <h2
        class="pt-5"
      >
        【未対応】
        4
        件
      </h2>
      <ul
        class="pl-5"
      >
        <li
          class="pt-2"
        >
          2023年10月01日
          までに対応必須の項目が
          2
          件あります
        </li>
      </ul>
      <ul
        class="pl-5"
      >
        <li
          class="pt-2"
        >
          2023年10月15日
          までに対応必須の項目が
          1
          件あります
        </li>
      </ul>
      <ul
        class="pl-5"
      >
        <li
          class="pt-2"
        >
          2023年12月15日
          までに対応必須の項目が
          1
          件あります
        </li>
      </ul>
    </div>
  </div>
</div>
`;
